presto-parser/src/main/java/com/facebook/presto/sql/parser/SqlParserOptions.java

```diff
@@ -13,14 +13,21 @@
  */
 package com.facebook.presto.sql.parser;
 
+import com.google.common.collect.ImmutableSet;
 import com.google.common.collect.Iterables;
 
 import java.util.EnumSet;
+import java.util.Set;
 
 import static java.util.Objects.requireNonNull;
 
 public class SqlParserOptions
 {
+    //TODO: discuss - not the most extendible way of maintaining such a list.
+    // Considerations: this list would change infrequently, and the warning is temporary.
+    protected static final Set<String> RESERVED_WORDS_WARNING = ImmutableSet.of(
+            "CALLED", "called", "DETERMINISTIC", "deterministic", "FUNCTION", "function", "LANGUAGE", "language", "RETURN", "return", "RETURNS", "returns", "SQL", "sql");
```

Comment:
upper case is good enough. Use `equalsIgnoreCase` to check

Reply:
Since this is a Set, it doesn't have ```containsIgnoreCase```. I could use upper case and then do something like ```set.contains(word.toUpperCase())``` as an alternative, but the tradeoff is that we would be paying the cost of ```toUpperCase``` everytime. I agree that would be cleaner but with a tiny performance overhead :)

Reply:
This is more about correctness than performance. `DeTeRMINIstic` is also a valid reserved word.

Reply:
good point. I didn't think of that :D

Reply:
You can also use `toUpperCase` or `toLowerCase` for comparison. As @highker said, this is about correctness. SQL keywords are not case sensitive.